### ISSUE TITLE
Documentation: Clarify _RelabelConfig_ usage in API documentation

### ISSUE THREAD

**Issue by opusmagnum** [coreos/prometheus-operator#1014]
**What did you do?**
Tried to relabel some metrics, adding `RelabelConfig` spec to the ServiceMonitor manifest.
**What did you expect to see?**
I have expected, that this spec will be transformed into `<relabel_configs>` section. 
**What did you see instead? Under which circumstances?**
A `<metric_relabel_configs>` section was generated by Prometheus Operator instead.
**Environment**

* Kubernetes version information:

```
Client Version: version.Info{Major:"1", Minor:"9", GitVersion:"v1.9.3", GitCommit:"d2835416544f298c919e2ead3be3d0864b52323b", GitTreeState:"clean", BuildDate:"2018-02-09T21:51:54Z", GoVersion:"go1.9.4", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"9", GitVersion:"v1.9.3+coreos.0", GitCommit:"f588569ed1bd4a6c986205dd0d7b04da4ab1a3b6", GitTreeState:"clean", BuildDate:"2018-02-10T01:42:55Z", GoVersion:"go1.9.2", Compiler:"gc", Platform:"linux/amd64"}
```

* Kubernetes cluster kind: 

Kubespray

* Manifests:

```
apiVersion: monitoring.coreos.com/v1
kind: ServiceMonitor
metadata:
  name: appds
  namespace: ns1
  labels:
    k8s-app: myapp
spec:
  endpoints:
    - port: http-metrics
      path: /metrics/prometheus
      # honorLabels: true
      interval: 30s
      # targetPort: 0
      metricRelabelings:
        - sourceLabels:	['__meta_kubernetes_pod_node_name']
          separator: ';'
          targetLabel: 'node_name'
          regex: '(.*)'
          replacement: '$1'
          action: replace
  jobLabel: my-app
  namespaceSelector:
    matchNames:
      - test
  selector:
    any: true
```

* Prometheus Operator Logs:

```n/a```

I have created a pull request with a suggested clarification.

**Comment by opusmagnum** [comment]
Do I undestand correctly, that `__meta_` labels are only available for [relabeling](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#relabel_config) but not for [metrics_relabeling](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#metric_relabel_configs)?

**Comment by opusmagnum** [comment]
Fixed with PR #1015.
Special thanks to @brancz for a great support and patience!
